### PR TITLE
Upgrade to node 20.12.22 & grunt-contrib-uglify to 5.2.2

### PR DESCRIPTION
**Pull request by Timodie**
### Description 
In a similar fashion to https://github.com/silvermine/videojs-airplay/pull/54, this PR upgrades the repo node 20.12.22 and also upgrades `"grunt-contrib-uglify` to `5.2.2`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,34 +5,59 @@ on: [ push, pull_request ]
...
-            uses: actions/setup-node@v3
+            uses: actions/setup-node@v4
             with:
                node-version-file: '.nvmrc'
-         - run: npm i -g npm@8.5.5
+         -
+            name: Put NVM version in output
+            id: makeNodeVersionOutput
+            run: echo "nvmrc=$(cat .nvmrc)" >> "$GITHUB_OUTPUT"
+
```
Comment (kmuncie) [inline .github/workflows/ci.yml:23]: is this new line intended?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -5,34 +5,59 @@ on: [ push, pull_request ]
...
+            fetch-depth: 0 # Fetch all history
          -
             name: Use Node.js ${{ matrix.node-version }}
-            uses: actions/setup-node@v3
+            uses: actions/setup-node@v4
             with:
                node-version: ${{ matrix.node-version }}
-         - run: npm i -g npm@8.5.5
          - run: npm ci # Reinstall the dependencies to ensure they install with the current version of node
          - run: npm test
```
Comment (kmuncie) [inline .github/workflows/ci.yml:48]: Should we be running the following before `npm test`?

```
         - run: npm run standards
         - run: npm run build # Ensure building is possible with this version of node
```

File: .nvmrc [hunk 3]
```diff
@@ -1 +1 @@
-16.15.0
+20.12.2
```
Comment (kmuncie) [inline .nvmrc:1]: I believe we are standardizing on including the `v` here

File: package.json [hunk 4]
```diff
@@ -5,15 +5,18 @@
   "main": "src/js/index.js",
   "scripts": {
     "commitlint": "commitlint --from 5ed6165",
-    "test": "check-node-version --npm 8.5.5 && nyc mocha -- -R spec 'tests/**/*.test.js'",
+    "check-node-version": "check-node-version --node $(cat .nvmrc) --npm 10.5.0 --print",
+    "test": "nyc mocha -- -R spec 'tests/**/*.test.js'",
```
Comment (kmuncie) [inline package.json:9]: I don't see the `-R` "reporter" option configured in many other projects. I know it was here before, but should it continue to be included?

**Review by kmuncie — COMMENTED** [review]
Just a first pass review, @joshuacurtiss can take it from here

**Comment by joshuacurtiss** [comment]
I concur with @kmuncie, and may I add a few more points? 

- Is it ok if we change the PR title and description to say `node 20.12.2`? Just to reduce confusion for anyone that looks back. Unfortunately the branch name can't be changed without complications to the PR, I think. 
- Is it ok if we have the commits that upgrade packages be done with actual `npm install`, which will accordingly update package-lock.json and help the commit be a complete piece of work?
- Whereas regenerating package-lock.json leaves us with the same result when we're all said and done, we regenerate package-lock.json separately because it's actually a different task with its own purpose (to get newest versions of all subdependencies), which also has its own handful of potential to introduce bugs. That's why I've found it meaningful to separate package-lock.json regeneration after doing specific package version upgrades.
- Just a question: On the grunt-contrib-uglify commit, did you encounter the bug mentioned in the commit description, or were you copying the message from my commit in the other repo? Just checking. If you did encounter the same error, great! The message is clear. If not, upgrading the package is perfect, but let's not talk about an error that isn't happening in this repo, if that makes sense..
- I just noticed that some of the commit message were really short line lengths. It's a minor thing but is it ok if we write out the messages in full lines?

**Comment by joshuacurtiss** [comment]
Please reference #179.

**Comment by onebytegone** [comment]
Superseded by https://github.com/silvermine/videojs-chromecast/pull/179

**Pull request closed without merging**